bahmni_api_feed/models/order_save_service.py

```diff
@@ -415,23 +411,23 @@ def _create_sale_order_line_function(self, sale_order, order):
                 else:
                     self._create_single_order_line(sale_order, prod_id, prod_obj, sorted_batches, actual_quantity, order_line_uom, description, order_line_dispensed, order)
 
-
     def _create_multiple_order_lines(self, sale_order, prod_id, prod_obj, sorted_batches, product_uom_qty, order_line_uom, description, order_line_dispensed, order):
         sale_order_line_obj = self.env['sale.order.line']
-        total_quantity_available = sum([lot['stock_forecast'] for _, lot in sorted_batches])
-        available_batches = [(lot_id, lot) for lot_id, lot in sorted_batches if lot['stock_forecast'] > 0]
+        # Calculate total_quantity_available from prodlot objects
+        total_quantity_available = sum([lot.quantity for lot in sorted_batches])
+        available_batches = [lot for lot in sorted_batches if lot.quantity > 0]  # List comprehension to filter available batches
```

Comment:
The filtering for batches with quantity > 0 seems redundant here as the get_available_batches function already returns only batches with quantity >0

Reply:
Refactored Mohan